### DISCUSSION TITLE
Failed to look up nuget package for some packages only

### DISCUSSION THREAD

**Discussion by svelue** [renovatebot/renovate#28788] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab 17.197.0 https://gitlab.com/renovate-bot/renovate-runner/-/raw/v17.197.0/templates/renovate.gitlab-ci.yml

### Please tell us more about your question or problem

I'm trying to configure renovate for a .net solution which has dependencies to public nuget.org + a private artifactory nuget feed.

I managed to configure everything, but for some reason some of the private packages could not be loaded while others can. They all are in the same private registry.

This is my config.js:

```
Object.assign(process.env, {
  GIT_AUTHOR_NAME: 'Renovate Bot',
  GIT_AUTHOR_EMAIL: 'bot@example.com',
  GIT_COMMITTER_NAME: 'Renovate Bot',
  GIT_COMMITTER_EMAIL: 'bot@example.com',
});

module.exports = {
  nuget: {
    registryUrls: [
	  "https://api.nuget.org/v3/index.json",
      "https://artifactory.xxx.com/artifactory/api/nuget/xxx-nuget-local"
    ]
  },
  hostRules: [
    {
      matchHost: 'https://artifactory.xxx.com/artifactory/api/nuget/xxx-nuget-local',
      username: process.env.ARTIFACTORY_xxx_USER,
      password: process.env.ARTIFACTORY_xxx_PASSWORD,
    },
  ],
  platform: 'gitlab',
  username: 'renovate-bot',
  gitAuthor: 'Renovate Bot <bot@example.com>',
  dependencyDashboard: true,
  extends: [
    "config:best-practices",
    "group:allNonMajor"
  ]
};
```

This is the error:
```
DEBUG: GET [https://artifactory.xxx.com:443/artifactory/api/nuget/xxx-nuget-local/FindPackagesById](https://artifactory.xxx.com/artifactory/api/nuget/xxx-nuget-local/FindPackagesById)()?%24select=Version%2CIsLatestVersion%2CProjectUrl%2CPublished&id=%27libXXX%27&%24skip=100 = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=401 retryCount=0, duration=42) (repository=ABC)
DEBUG: Datasource unauthorized (repository=ABC)
       "datasource": "nuget",
       "packageName": "libXXX",
       "url": "https://artifactory.xxx.com/artifactory/api/nuget/xxx-nuget-local/FindPackagesById()?%24select=Version%2CIsLatestVersion%2CProjectUrl%2CPublished&id=%27libXXX%27&%24skip=100"
DEBUG: Failed to look up nuget package libXXX (repository=ABC, packageFile=some.csproj, dependency=libXXX)
```

It's always the same 5 packages which could not be loaded (out of 8-10 from the xxx-nuget-local) and I already tried to set concurrentRequestLimit and maxRequestsPerSecond to 2 in that hostRule.

Any ideas??

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
If you disable the working packages using packageRules then do the remaining ones all still fail?

**Reply by svelue** [reply]
I added 
```
  packageRules: [
	{
    excludePackageNames: [
	  "lib1",
	  "lib2",
	  "lib3",
	  "lib4",
	  "lib5",
	  "lib6",
	],	
	}
  ],
```
to the module.exports and see no changes. These are the exact names of the dependencies which are successfully loaded from that private registry. The same 5 other ones are still failing.
I'm confused because the packageRules object in the logged combined config is still empty?? Is that correct? On the other hand I'm sure the config is loaded because at first I had a syntax error in it ;)

**Reply by rarkins** [reply]
No, that rule doesn't do anything. You need to match (not exclude) and then set enabled=false

**Reply by svelue** [reply]
Ok, so now the previously working packages are disabled but the other ones are still not working :/

**Reply by rarkins** [reply]
That probably rules out the concern that it's anything to do with rate limiting.

Next step: can you reduce it down to a single dependency, and it fails to look up? No need to debug 5+ when you only need one.

And if so, can you then have one working and one non-working, which are as similar to each other as possible?

**Reply by svelue** [reply]
Sorry for that late reply but I was on vacation and had some other stuff to do.

I tried your suggestions but the 5 libs are still failing. Actually I can't imagine what might be different between the libs, they are all build using the same ci scripts and pushed into the same artifactory nuget space.

I also tried to get the versions manually via https://artifactory.xxx.com/artifactory/api/nuget/v3/xxx-nuget-local/query?q=libXXX and I'm getting all versions, as expected, for both, one of the working and one of the buggy libs. Also I tried this with my personal and with the ci credentials provided to renovate.
What endpoint are u using in renovate? Maybe I can try to query that manually?

**Reply by rarkins** [reply]
Renovate defaults to the public NuGet registry

**Reply by svelue** [reply]
But when providing a private url via nuget.registryUrls renovate should check this registry too? Otherwise it wouldn't work, right?^^

**Reply by rarkins** [reply]
If you override the urls, they're fully overridden. If you want to include the public registry too, you need to add it. Some people don't want the public registry used at all.

**Reply by svelue** [reply]
With the above configuration it should use both registries, right?

**Reply by rarkins** [reply]
Not in all cases. Use packageRules instead to be sure
